addons/website_slides/static/src/js/public/components/slide_upload_dialog/slide_upload_category.js

```diff
@@ -0,0 +1,441 @@
+            } else {
+                this.state.preview.show = true;
+            }
+        }
+
+        this.state.form.isLoading = false;
+    }
+
+    async onClickFormSubmit(forcePublished) {
+        if (!this._formValidate()) {
+            return;
+        }
+        const values = await this._formValidateGetValues(forcePublished);
+        this.props.upload(values, this.props.slideCategory);
+    }
+
+    /**
+     * When the user selects 'local_file' or 'external' as source type, we display the 'upload'
+     * field or the 'document_google_url' / 'image_google_url' fields respectively.
+     */
+    onClickSourceType(isLocalSource) {
+        this.state.form.isLocalSource = isLocalSource;
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    // Alert messages
+
+    /**
+     * @param {string} message
+     */
+    _alertDisplay(message, alertClass = "alert-warning") {
+        this.state.alert.message = message;
+        this.state.alert.class = alertClass;
+        this.state.alert.show = true;
+    }
+
+    _alertRemove() {
+        this.state.alert.show = false;
+        this.state.alert.message = "";
+        this.state.alert.class = "";
+    }
+
+    // Category and tag SelectMenus
+
+    /**
+     * Get value for category_id and tag_ids (ORM cmd) to send to server
+     */
+    _getSelectMenuValues() {
+        const result = {};
+        // tags
+        if (this.state.choices.tagIds.length > 0) {
+            result["tag_ids"] = this.state.choices.tagIds.map((tagId) => {
+                const tag = this.state.choices.tags.find((tag) => tag.value === tagId);
+
+                if (this._toCreate(tag.value)) {
+                    return [0, 0, { name: tag.label }];
+                } else {
+                    return [4, tag.value];
+                }
+            });
+        }
+        // category
+        if (!this.defaultCategoryId) {
+            if (this._toCreate(this.state.choices.categoryId)) {
+                const category = this.state.choices.categories.find((cat) => {
+                    return cat.value === this.state.choices.categoryId;
+                });
+                result["category_id"] = [0, { name: category.label }];
+            } else {
+                const categoryId = parseInt(this.state.choices.categoryId, 10);
+                result["category_id"] = [categoryId];
+                this.categoryID = categoryId;
+            }
+        } else {
+            result["category_id"] = [this.defaultCategoryId];
+        }
+        return result;
+    }
+
+    /**
+     * Fetch available course categories and tags
+     */
+    async _fetch_choices(type, domain = [], fields = ["name"]) {
+        const choices = [];
+        const results = await rpc(`/slides/${type}/search_read`, { fields, domain });
+
+        results.read_results.forEach((choice) => {
```

Comment:
Look like some comments of the previous review are missing...

can you check if other comment are missing ?

![image](https://github.com/odoo/odoo/assets/54316226/938184e6-9083-401e-b689-c572b9c41cfb)